book/src/chap18.asciidoc

```diff
@@ -27,590 +27,590 @@ For example, this table shows the suits and the corresponding integer codes:
+Carta(2, 12)
 ----
 
 
-=== Global Variables
+=== Variáveis Globais
 
-In order to print +Card+ objects in a way that people can easily read, we need a mapping from the integer codes to the corresponding ranks and suits. A natural way to do that is with arrays of strings:
+Para imprimir objetos +Carta+ de uma forma que as pessoas possam ler facilmente, precisamos de um mapeamento dos códigos inteiros até os valores e naipes correspondentes. Uma maneira natural de fazer isso é com arrays de strings:
 
 [source,@julia-setup chap18]
 ----
-const suit_names = ["♣", "♦", "♥", "♠"]
-const rank_names = ["A", "2", "3", "4", "5", "6", "7", "8", "9", "10", "J", "Q", "K"]
+const nomes_naipes = ["♣", "♦", "♥", "♠"]
+const nomes_valores = ["A", "2", "3", "4", "5", "6", "7", "8", "9", "10", "J", "Q", "K"]
 ----
 
-The variables +suit_names+ and +rank_names+ are global variables. The +const+ declaration means that the variable can only be assigned once. This solves the performance problem of global variables.
-(((global statement)))(((const)))((("keyword", "const", see="const")))
+As variáveis +nomes_naipes+ e +nomes_valores+ são variáveis globais. A declaração +const+ significa que a variável só pode ser atribuída uma vez. Isso resolve o problema de desempenho das variáveis globais.
+(((declaração global)))(((const)))((("palavra-chave", "const", see="const")))
 
-Now we can implement an appropriate +show+ method:
+Agora podemos implementar um método +show+ apropriado:
 (((show)))
 
 [source,@julia-setup chap18]
 ----
-function Base.show(io::IO, card::Card)
-    print(io, rank_names[card.rank], suit_names[card.suit])
+function Base.show(io::IO, carta::Carta)
+    print(io, nomes_valores[carta.valor], nomes_naipes[carta.naipe])
 end
 ----
 
-The expression +rank_names[card.rank]+ means “use the field +rank+ from the object +card+ as an index into the array +rank_names+, and select the appropriate string.”
+A expressão +nomes_valores[carta.valor]+ significa “use o campo +valor+ do objeto +carta+ como um índice na matriz +nomes_valores+, e selecione a string apropriada.”
 
-With the methods we have so far, we can create and print cards:
+Com os métodos que temos até agora, podemos criar e imprimir as cartas:
 
 [source,@julia-repl-test chap18]
 ----
-julia> Card(3, 11)
+julia> Carta(3, 11)
 J♥
 ----
 
 
-=== Comparing Cards
+=== Comparando Cartas
 
-For built-in types, there are relational operators (+<+, +>+, +==+, etc.) that compare values and determine when one is greater than, less than, or equal to another. For programmer-defined types, we can override the behavior of the built-in operators by providing a method named: +<+.
+Para os tipos embutidos, existem operadores relacionais (+<+, +>+, +==+ etc.) que comparam os valores e determinam quando um é maior que, menor que ou igual ao outro. Para os tipos definidos pelo programador, podemos sobrepor o comportamento dos operadores embutidos ao estabelecer um método chamado +<+.
```

Comment:
Veja se concorda, traduzir "built-in types" para "tipos internos"
```suggestion
Para os tipos internos, existem operadores relacionais (+<+, +>+, +==+ etc.) que comparam os valores e determinam quando um é maior que, menor que ou igual ao outro. Para os tipos definidos pelo programador, podemos sobrepor o comportamento dos operadores internos ao estabelecer um método chamado +<+.
```

Reply:
Tanto faz traduzir para embutido ou interno. Durante a tradução, os dois termos foram usados.